main.c

```diff
@@ -122,17 +122,51 @@
 uint8_t value_out = 0;
 #endif
 
-#if MICROPY_ENABLE_PYSTACK
-static size_t PLACE_IN_DTCM_BSS(_pystack[CIRCUITPY_PYSTACK_SIZE / sizeof(size_t)]);
+#if MICROPY_ENABLE_PYSTACK && CIRCUITPY_OS_GETENV
+#include "shared-module/os/__init__.h"
 #endif
 
+typedef struct {
+    #if MICROPY_ENABLE_PYSTACK
+    supervisor_allocation *pystack;
+    #endif
+    supervisor_allocation *heap;
+} stacks;
```

Comment:
I think we could come up with a better name for this. I suggest:
stacks -> vm_memory_t (the _t suffix is the convention for typedefs in C)
alloc_stacks -> allocate_vm_memory
combo -> vm_memory

Reply:
Yea these names are much better, doing it.